tests/internal/test_settings.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+
+from ddtrace import config
+
+
+# Reset the configuration between each test case.
+# This is necessary because the config object is a singleton.
+@pytest.fixture(autouse=True)
+def reset_config():
+    config.reset()
+    yield
+    config.reset()
+
+
+@pytest.mark.parametrize(
+    "test",
+    [
+        {
+            "expected_service": None,
+        },
+        {
+            "env": {"DD_SERVICE": "env_svc"},
+            "expected_service": "env_svc",
+        },
+        {
+            "env": {"DD_TAGS": "service:env_svc"},
+            "expected_service": "env_svc",
+        },
+        {
+            "code": "code_svc",
+            "expected_service": "code_svc",
+        },
+        {
+            "env": {"DD_SERVICE": "env_svc1", "DD_TAGS": "service:env_svc2"},
+            "expected_service": "env_svc1",
+        },
+        {
+            "env": {"DD_SERVICE": "env_svc1", "DD_TAGS": "service:env_svc2"},
+            "code": "code_svc",
+            "expected_service": "code_svc",
+        },
+    ],
+)
+def test_service(test, monkeypatch):
```

Comment:
Should we add a test case that verifies `tracer._service` caches the expected value?